File: backend/AspireApp/AspireApp.ApiService/Models/Order.cs
namespace AspireApp.ApiService.Models;

public class Order
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public User User { get; set; } = null!;
    public string Status { get; set; } = "Pending"; // Pending, Processing, Shipped, Delivered, Cancelled
    public decimal TotalAmount { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public DateTime? ShippedAt { get; set; }
    public DateTime? DeliveredAt { get; set; }
    public List<OrderItem> OrderItems { get; set; } = new();
    public string ShippingAddress { get; set; } = string.Empty;
    public string? TrackingNumber { get; set; }
} 

File: frontend/src/types/order.ts
export interface OrderItem {
  id: string
  productId: string
  productName: string
  quantity: number
  price: number
}

export interface Order {
  id: string
  customerName: string
  customerEmail: string
  state: 'Pending' | 'Processing' | 'Shipped' | 'Delivered' | 'Cancelled'
  totalAmount: number
  createdAt: string
  items: OrderItem[]
} 